### PR TITLE
upgrade tendermint to v0.33.0

### LINKED ISSUE #1332: Upgrade to Tendermint v0.33.1
**Is your feature request related to a problem? Please describe.**

The recent Tendermint upgrade (to [v0.33.1](https://github.com/tendermint/tendermint/blob/master/CHANGELOG.md#v033)) involves a large number of backward-incompatible changes. Due to Go Module's weak capability in handling dependency hell problem of `v0` packages, related projects may be influenced.

**Describe the solution you'd like**

Upgrade to Tendermint `v0.33.1`.

**Describe alternatives you've considered**

Adding a `replace` rule in `go.mod` that redirects `tendermint` to `tendermint v0.32.x`.

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Closes #1332

### CODE REVIEW

File: bcm/blockchain.go [hunk 1]
```diff
@@ -29,10 +29,12 @@ type BlockchainInfo interface {
 	LastCommitDuration() time.Duration
 	LastBlockHash() []byte
 	AppHashAfterLastBlock() []byte
-	// Gets the BlockHash at a height (or nil if no BlockStore mounted or block could not be found)
+	// BlockHash gets the hash at a height (or nil if no BlockStore mounted or block could not be found)
 	BlockHash(height uint64) ([]byte, error)
```
Comment (silasdavis) [inline bcm/blockchain.go]: Remove this - it is unused now

**Review by silasdavis — CHANGES_REQUESTED** [review]
Looks good!

Please remove `BlockHash` function from `Blockchain`

Might as well bump to 0.33.1 now. Will merge once rebased.

File: bcm/blockchain.go (hunk 1 above)
Comment (gregdhill) [reply bcm/blockchain.go]: It is used here > https://github.com/hyperledger/burrow/blob/master/execution/evm/contract.go#L464

Comment (gregdhill) [reply bcm/blockchain.go]: It is still used here > https://github.com/hyperledger/burrow/blob/master/execution/evm/contract.go#L464

**Pull request merged**